.github/workflows/pr-preview.yml

```diff
@@ -76,6 +76,9 @@ jobs:
             });
 
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
In case anyone else is wondering, even though the trigger is `pull_request_target`, the event payload is `pull_request` according to the [docs](https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#pull_request_target).